src/Tribe/API.php

```diff
@@ -602,5 +642,222 @@ public static function get_and_flatten_event_meta( $event_id ) {
 
 			return $post_meta;
 		}
+
+		/**
+		 * Sanitize the arguments array before sending to create/update an event post.
+		 *
+		 * Use this prior to sending arguments to post create/update function.
+		 *
+		 * @since TBD
+		 *
+		 * @see Tribe__Events__API::sanitize_meridian_meta_value
+		 * @see Tribe__Events__API::sanitize_hour_meta_value
+		 * @see Tribe__Events__API::sanitize_minute_meta_value
+		 *
+		 * @param array $args The arguments sent to create/update an event post.
+		 *
+		 * @return array|WP_Error
+		 */
+		public static function sanitize_event_post_create_update_args( $args ) {
+			if (
+				empty( $args['post_type'] )
+				|| Tribe__Events__Main::POSTTYPE !== $args['post_type']
+				|| ! is_array( $args )
+			) {
+				return $args;
+			}
+
+			// Sanitize if valid, or fail with WP_Error if invalid.
+			// Could enhance this with more comprehensive checks in the future.
+
+			// Process meridian fields before hour fields to determine if hours are to be in the 12-hour or 24-hour format.
+			if ( ! empty( $args['EventStartMeridian'] ) ) {
+				$args['EventStartMeridian'] = self::sanitize_meridian_meta_value( $args['EventStartMeridian'] );
+			}
+
+			if ( ! empty( $args['EventEndMeridian'] ) ) {
+				$args['EventEndMeridian'] = self::sanitize_meridian_meta_value( $args['EventEndMeridian'] );
+			}
+
+			// If meridian is set but we can pretty easily guess the hour is a valid 24-hour format, discard meridian in attempt to be smarter/flexible, thus setting "14pm" (invalid) to "14" (equivalent to 2pm).
+			// We take this approach instead of just minus `12` from the integer value of the hour because meridian may be set by defaults.
+			// Watch out for sending "12pm" or "12am" as the meridian (12-hour format) will take precedence if it exists.
+			if (
+				! empty( $args['EventStartMeridian'] )
+				&& 12 < absint( $args['EventStartHour'] )
+				&& 24 > absint( $args['EventStartHour'] )
+			) {
+				$args['EventStartMeridian'] = '';
+			}
+
+			if (
+				! empty( $args['EventEndMeridian'] )
+				&& 12 < absint( $args['EventEndHour'] )
+				&& 24 > absint( $args['EventEndHour'] )
+			) {
+				$args['EventEndMeridian'] = '';
+			}
+
+			// Now process all but the meridians
+			foreach ( $args as $key => &$value ) {
+				if ( 'EventStartHour' === $key ) {
```

Comment:
To avoid deep nesting you can return here the args instead.

```php
foreach( ... ) {
	if ( 'EventStartHour' !== $key ) {
		return $args;
	}
	
	// Remove nesting and do other operations
}
```

Reply:
`$key` could be one of four different meta keys, not just this one, right?

Reply:
Right anything that is not `EventStartHour`

Reply:
discussed via Slack and Cris said leave as-is